src/cascadia/TerminalApp/App.cpp

```diff
@@ -843,17 +823,26 @@ namespace winrt::TerminalApp::implementation
     // - tabViewItem: the TabViewItem in the TabView that is being removed.
     void App::_RemoveTabViewItem(const IInspectable& tabViewItem)
     {
+        // TODO: GitHub:627: Need a better story for what should happen when the last tab is closed.
+        if (_tabs.size() <= 1) {
+            return;
+        }
+
         uint32_t tabIndexFromControl = 0;
         _tabView.Items().IndexOf(tabViewItem, tabIndexFromControl);
 
         if (tabIndexFromControl == _GetFocusedTabIndex())
         {
+            // switch to the "future" tab first, or current tab will not be controlled by tabview but remain focused and shown

```

Comment:
I don't think I understand what this comment is trying to say - could you clarify?

Reply:
My poor expression... will remove this comment